packages/Sensio/src/Rector/FrameworkExtraBundle/TemplateAnnotationRector.php

```diff
@@ -106,6 +113,17 @@ public function refactor(Node $node): ?Node
         return $node;
     }
 
+    private function addBaseClassIfMissing(Node $node): ?Node
+    {
+        if ($node->extends !== null) {
+            return null;
+        }
+
```

Comment:
Not all classes need to be extended.
This would add parent controller to every single class, that has no parent yet.

To fix that, there must be check that at least one of class' method has the template annotation.
If not, return null here

Reply:
Good point. I updated the PR.

Reply:
Very nice job!